05_iigo/index.tex

```diff
@@ -547,11 +547,34 @@ \subsubsection{Server-side}
     \item The required information from agents or other roles is obtained. This information includes \emph{allocation requests}, \emph{private pool resource reports} and \emph{rules} proposed by the islands.
     \item This information is aggregated into a format accepted by the interface of presidential functions (usually in form $map[agent] \xrightarrow[]{} information$).
     \item Presidential function is called only on the currently residing President with aggregated information as inputs.
-    \item The outcome of the presidential function call is processed.
+    \item The outcome of the presidential function call is processed. This includes updating the game state, logging information for monitoring, subtracting action costs.
     \item The outcome of the action, or lack thereof, is passed to the members of the archipelago.
 \end{enumerate}
 
-\subsubsection{Legislative Branch}
+\subsection{Legislative Branch}
+Similarly to the president the Speaker also has client-side functions which directly map to the Speaker powers described in \ref{sec:legislative}, as well as server-side functions which not only facilitate the implementation of these powers but also handle other functionality, such as the logging needed for monitoring and withdrawal of costs associated with the power.
+
+\subsubsection{Client-side}
+The legislative branch functions can be divided into two categories:
+\begin{enumerate}
+\item Speaker power functions:
+    \begin{itemize}
+        \item \texttt{DecideAgenda} does not directly map to a power of the speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, they can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
```

Comment:
```suggestion
        \item \texttt{DecideAgenda} does not directly map to the power of the Speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, they can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
```
who is 'they' in the last sentence?? can you plz be more explicit :)

Reply:
is it 'the Speaker'

Reply:
```suggestion
        \item \texttt{DecideAgenda} does not directly map to any power of the Speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, agents can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
```